Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -0,0 +1,835 @@
+                    hourAction = switch frequency {
+                    case .minutely, .hourly: .limit
+                    case .yearly, .monthly, .weekly, .daily: .expand
+                    }
+                }
+                
+                if recurrence.minutes.isEmpty { 
+                    minuteAction = nil
+                } else {
+                    minuteAction = switch frequency {
+                    case .minutely: .limit
+                    case .yearly, .monthly, .weekly, .daily, .hourly: .expand
+                    }
+                }
+                
+                if recurrence.seconds.isEmpty { 
+                    secondAction = nil
+                } else {
+                    secondAction = .expand
+                }
+                
+                if let range = range {
+                    rangeLowerBoundInterval = recurrence.calendar.dateInterval(of: frequency.component, for: range.lowerBound) 
+                } else {
+                    rangeLowerBoundInterval = nil
+                }
+                
+                // Create date components that enumerate recurrences without any
+                // rules applied. Retrieve the date components of the start date
+                // but leave the field for the recurrence frequency empty, so it
+                // is the only component that increases
+#if FIXED_124640028
+                // rdar://124640028
+                // https://github.com/apple/swift-foundation/issues/483
+                // Calendar.dates(byMatching: ...) does not respect repeatedTimePolicy when .nanosecond is set #483
+                let components: Calendar.ComponentSet = switch recurrence.frequency {
+                    case .minutely: [.nanosecond, .second]
+                    case .hourly:   [.nanosecond, .second, .minute]
+                    case .daily:    [.nanosecond, .second, .minute, .hour]
+                    case .weekly:   [.nanosecond, .second, .minute, .hour, .weekday]
+                    case .monthly:  [.nanosecond, .second, .minute, .hour, .day]
+                    case .yearly:   [.nanosecond, .second, .minute, .hour, .day, .month]
+                }
+#else
+                let components: Calendar.ComponentSet = switch recurrence.frequency {
+                    case .minutely: [.second]
+                    case .hourly:   [.second, .minute]
+                    case .daily:    [.second, .minute, .hour]
+                    case .weekly:   [.second, .minute, .hour, .weekday]
+                    case .monthly:  [.second, .minute, .hour, .day]
+                    case .yearly:   [.second, .minute, .hour, .day, .month]
+                }
+#endif
+                let componentsForEnumerating = recurrence.calendar._dateComponents(components, from: start) 
+                
+                let rangeForBaseRecurrence: Range<Date>? = nil
+                baseRecurrence = Calendar.DatesByMatching(calendar: recurrence.calendar,
```

Comment:
Why create this directly instead of calling the `dates(byMatching:...)` API on `Calendar`? There doesn't seem to be much reason to break the abstraction.

Reply:
We need to store the iterator, since it is used throughout the enumeration and we don't want to reset. We can't use `dates(byMatching: ...).makeIterator()` because we need to save this in a var the structure, and thus it'd need to be `Sendable`. `any (IteratorProtocol<Date> & Sendable)` does not currently compile (rdar://104888113), so we just represent it using the internal type which is known to be `Sendable`